weatherApp/Controllers/HomeViewController.swift

```diff
@@ -9,9 +9,138 @@
+    private var currentWeather: [CurrentWeather] = []
+    private var refreshControl: UIRefreshControl!
+    
+    @IBOutlet private weak var tableView: UITableView!
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         
+        setupNavigationBar()
+        setupTableView()
+        setupData()
+    }
+    
+    override func viewWillLayoutSubviews() {
+        super.viewWillLayoutSubviews()
+
+        view.setGradientBackground(colorOne: Colors.grey, colorTwo: Colors.darkNavyBlue)
+    }
+    
+    override func willTransition(to newCollection: UITraitCollection, with coordinator: UIViewControllerTransitionCoordinator) {
+        refreshTableView()
+    }
+    
+    //MARK: - Data
+    
+    private func setupData() {
+        currentWeather = []
+        
+        for city in Cities.allCases {
+            WeatherService().fetchCurrentWeather(id: city.rawValue) { [weak self] (currentWeather) in
+                guard
+                    let self = self,
+                    let currentWeather = currentWeather else { return }
+                self.currentWeather.append(currentWeather)
+                self.refreshTableView()
+            }
+        }
+    }
+    
+    private func numberOfCurrentWeather() -> Int {
+        return currentWeather.count
+    }
+    
+    private func currentWeather(atIndex index: Int) -> CurrentWeather? {
+        guard currentWeather.count > index else { return nil }
+        
+        return currentWeather[index]
+    }
+    
+    //MARK: - UI elements setup
+    
+    private func setupNavigationBar(){
+        title = "WeatherApp"
+        
+        navigationController?.navigationBar.prefersLargeTitles = true
+        navigationItem.backBarButtonItem = UIBarButtonItem(title: "Back", style: .plain, target: nil, action: nil)
+        navigationController?.navigationBar.tintColor = UIColor.white;
+        navigationController?.navigationBar.largeTitleTextAttributes = [NSAttributedString.Key.foregroundColor : UIColor.white]
+
+    }
+    
+    private func setupTableView() {
+        tableView.delegate = self
+        tableView.dataSource = self
+        
+        tableView.separatorStyle = .none
+        
+        refreshControl = UIRefreshControl()
+        refreshControl.addTarget(self, action: #selector(refreshTableView), for: UIControl.Event.valueChanged)
+        tableView.refreshControl = refreshControl
+        
+        tableView.register(UINib(nibName: "WeatherTableViewCell", bundle: nil), forCellReuseIdentifier: cellReuseIdentifier)
+    }
+    
+    @objc private func refreshTableView() {
+        DispatchQueue.main.async {
+            self.tableView.reloadData()
+            self.refreshControl.endRefreshing()
+        }
+    }
+    
+}
+
+
+//MARK: - TableView DataSource
+
+extension HomeViewController: UITableViewDataSource {
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return numberOfCurrentWeather()
```

Comment:
Since `numberOfCurrentWeather` is used only here, you could move the content of that `func` body here.